src/imars3d/backend/diagnostics/rotation.py

```diff
@@ -96,8 +96,12 @@ def _find_rotation_center(
             logger.error(msg)
             raise ValueError(msg)
         # locate 180 degree pairs
-        atol = atol_deg if in_degrees else np.radians(atol_deg)
-        idx_low, idx_hgh = find_180_deg_pairs_idx(angles, atol=atol, in_degrees=in_degrees)
+        if atol_deg is None:
```

Comment:
This diff can be removed (BTW: you forgot to convert units if it is guessing `atol`). Since `find_180_deg_pairs_idx` treats `atol=None` the same way as `find_rotation_center` so you don't have to branch.

Reply:
units conversion of angles is handled in find_180_deg_pairs_idx, which is why we need to pass in `in_degree`.